x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/execution_log_table/execution_log_table.tsx

```diff
@@ -68,28 +88,72 @@ const ExecutionLogTableComponent: React.FC<ExecutionLogTableProps> = ({
     storage,
     timelines,
   } = useKibana().services;
-  // Datepicker state
-  const [recentlyUsedRanges, setRecentlyUsedRanges] = useState<DurationRange[]>([]);
-  const [refreshInterval, setRefreshInterval] = useState(1000);
-  const [isPaused, setIsPaused] = useState(true);
-  const [start, setStart] = useState('now-24h');
-  const [end, setEnd] = useState('now');
 
-  // Searchbar/Filter/Settings state
-  const [queryText, setQueryText] = useState('');
-  const [statusFilters, setStatusFilters] = useState<string | undefined>(undefined);
-  const [showMetricColumns, setShowMetricColumns] = useState<boolean>(
-    storage.get(RULE_DETAILS_EXECUTION_LOG_TABLE_SHOW_METRIC_COLUMNS_STORAGE_KEY) ?? false
-  );
+  const {
+    [RuleDetailTabs.executionLogs]: {
+      state: {
+        superDatePicker: { recentlyUsedRanges, refreshInterval, isPaused, start, end },
+        queryText,
+        statusFilters,
+        showMetricColumns,
+        pagination: { pageIndex, pageSize },
+        sort: { sortField, sortDirection },
+      },
+      actions: {
+        setEnd,
+        setIsPaused,
+        setPageIndex,
+        setPageSize,
+        setQueryText,
+        setRecentlyUsedRanges,
+        setRefreshInterval,
+        setShowMetricColumns,
+        setSortDirection,
+        setSortField,
+        setStart,
+        setStatusFilters,
+      },
+    },
+  } = useRuleDetailsContext();
 
-  // Pagination state
-  const [pageIndex, setPageIndex] = useState(1);
-  const [pageSize, setPageSize] = useState(5);
-  const [sortField, setSortField] = useState<keyof AggregateRuleExecutionEvent>('timestamp');
-  const [sortDirection, setSortDirection] = useState<SortOrder>('desc');
   // Index for `add filter` action and toasts for errors
   const { indexPattern } = useSourcererDataView(SourcererScopeName.detections);
-  const { addError, addSuccess } = useAppToasts();
+  const { addError, addSuccess, remove } = useAppToasts();
+
+  // QueryString, Filters, and TimeRange state
+  const dispatch = useDispatch();
+  const getGlobalFiltersQuerySelector = useMemo(
+    () => inputsSelectors.globalFiltersQuerySelector(),
+    []
+  );
+  const getGlobalQuerySelector = useMemo(() => inputsSelectors.globalQuerySelector(), []);
+  const timerange = useDeepEqualSelector(inputsSelectors.globalTimeRangeSelector);
+  const query = useDeepEqualSelector(getGlobalQuerySelector);
+  const filters = useDeepEqualSelector(getGlobalFiltersQuerySelector);
+  const cachedGlobalQueryState = useRef<CachedGlobalQueryState>({ filters, query, timerange });
+  const successToastId = useRef('');
+
+  const resetGlobalQueryState = useCallback(() => {
+    dispatch(
+      // TODO: Determine relative vs absolute so `Today` is replaced instead of absolute
```

Comment:
Do we want to fix it in this PR or a follow-up? I didn't notice that the filter gets reset to absolute time and was confused about alerts not showing up. It took some time to figure out what happened 🙂

Reply:
Yeah, that's no good! Resolved in [22cc0c8](https://github.com/elastic/kibana/pull/130072/commits/22cc0c8dbd2a1300675caf4c6d471d211ed44858), was an easy conditional check 🙂